map/src/drawer/components/tracks/CloudTrackItem.jsx

```diff
@@ -31,35 +31,34 @@ export default function CloudTrackItem({ file }) {
     async function addTrackToMap(setProgressVisible) {
         setProgressVisible(true);
         if (file.url) {
-            ctx.setSelectedGpxFile(ctx.gpxFiles[file.name]);
+            ctx.setSelectedGpxFile(Object.assign({}, ctx.gpxFiles[file.name]));
         } else {
-            let url = `${process.env.REACT_APP_USER_API_SITE}/mapapi/download-file?type=${encodeURIComponent(
-                file.type
-            )}&name=${encodeURIComponent(file.name)}`;
+            const URL = `${process.env.REACT_APP_USER_API_SITE}/mapapi/download-file`;
+            const qs = `?type=${encodeURIComponent(file.type)}&name=${encodeURIComponent(file.name)}`;
             const newGpxFiles = Object.assign({}, ctx.gpxFiles);
             newGpxFiles[file.name] = {
-                url: url,
+                url: URL + qs,
                 clienttimems: file.clienttimems,
                 updatetimems: file.updatetimems,
                 name: file.name,
                 type: 'GPX',
             };
-            let f = await Utils.getFileData(newGpxFiles[file.name]);
+            const f = await Utils.getFileData(newGpxFiles[file.name]);
             const gpxfile = new File([f], file.name, {
                 type: 'text/plain',
             });
-            let track = await TracksManager.getTrackData(gpxfile);
+            const track = await TracksManager.getTrackData(gpxfile);
             setProgressVisible(false);
-            if (track) {
-                let type = ctx.OBJECT_TYPE_CLOUD_TRACK;
+            if (track && (track.points?.length > 0 || track.wpts?.length > 0 || track.tracks?.length > 0)) {
```

Comment:
this is a bad pattern, if someone will add new field that method won't work and nobody is going to fix. It's unclear by the code why empty points  track is not accepted which is not bad probably, and anyway such checks should be part of track object itself to make it more encapsulated like isEmpty method however I would question if it's really needed.

Reply:
TracksManager.isEmptyTrack was added